src/modules/rlm_pap/rlm_pap.c

```diff
@@ -375,11 +401,19 @@ static rlm_rcode_t CC_HINT(nonnull) mod_authorize(void *instance, REQUEST *reque
 
 #ifdef HAVE_OPENSSL_EVP_H
 		case PW_SHA2_PASSWORD:
+		case PW_SSHA224_PASSWORD:
+		case PW_SSHA256_PASSWORD:
+		case PW_SSHA384_PASSWORD:
+		case PW_SSHA512_PASSWORD:
```

Comment:
You're setting the minimum size to 28 for all SHA2 password types... That's my issue with this code. It will make it significantly more likely that the heuristics will misidentify the encoding of the "known good" password.